R/modules.R

```diff
@@ -212,7 +212,18 @@ module <- function(label = "module",
   }
 
   if (label == "global_filters") {
-    stop("Label 'global_filters' is reserved in teal. Please change to something else.")
+    stop(
+      sprintf("module(label = \"%s\", ...\n  ", label),
+      "Label 'global_filters' is reserved in teal. Please change to something else.",
+      call. = FALSE
+    )
+  }
+  if (label == "Report previewer") {
+    stop(
+      sprintf("module(label = \"%s\", ...\n  ", label),
+      "Label 'Report previewer' is reserved in teal.",
+      call. = FALSE
+    )
```

Comment:
```suggestion
  if (label %in% c("global_filters", "Report previewer")) {
    stop(
      sprintf(
        "module(label = %1$s, ...
        Label %1$s is reserved in teal. Please change to something else.
        ",
        label
      ),
      call. = FALSE
    )
  }
```

Perhaps we can combine the code for optimization?

Reply:
The line break within the `fmt` string adds loads of white space to the message, you should do 
```
    sprintf(
      "module(label = \"%s\", ...\n  %s",
      label,
      "Label 'global_filters' is reserved in teal. Please change to something else."
    )
```

The benchmark is exactly the same, down to a microsecond.
```
Unit: microseconds
    expr    min      lq     mean  median     uq     max neval cld
     new 36.793 41.2085 51.34256 44.1210 52.019 275.206   500   a
 current 36.873 41.2010 51.51580 44.2335 50.337 289.881   500   a
```

The current is 3 lines less.